packager/mux/mux.go

```diff
@@ -149,18 +161,13 @@ func (this *mux) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 
 	errorMsg := ""
 	errorCode := 0
-	if matchingRule == nil {
-		// Should never happen, because the last rule matches any path.
-		errorMsg, errorCode = "500 internal error", http.StatusInternalServerError
+	// Validate HTTP method and params, parse params and attach them to req.
+	if !allowedMethods[req.Method] {
+		errorMsg, errorCode = "405 method not allowed", http.StatusMethodNotAllowed
 	} else {
```

Comment:
Removed this because 1) there's a reference to matchingRule below that would crash if it was nil, so this check is not sufficient to gracefully handle this hypothetical situation 2) unit tests already verify that matchingRule won't be nil 3) an assert would be nice for readability, but it's probably OK to just delete this. I'll add a comment.